pilot/pkg/model/push_context.go

```diff
@@ -167,6 +167,9 @@ type PushContext struct {
 	// clusterLocalHosts extracted from the MeshConfig
 	clusterLocalHosts ClusterLocalHosts
 
+	// discoverabilityPolicy provides a global policy for controlling service discoverability across the mesh.
+	ServiceDiscoverabilityPolicy
```

Comment:
Same - this feature is great, but the fact it's leaking so deep into model and core interfaces ( and leaking impl details that will likely need to change ) is not great.